### PR TITLE
Quick lya

### PR DESCRIPTION
**Pull request by julienguy**
Script to generate spectra files from lya mock skewers. This is related to issue #338 .
For each skewer file (a healpix), 
 - skewers are read with desisim.lya_spectra.read_lya_skewers
 - quasar are simulated with desisim.templates.SIMQSO
 - lya transmission is applied with desisim.lya_spectra.apply_lya_transmission
 - DESI observations are simulated with desisim.scripts.quickspectra.sim_spectra 
 - optionally can write a fake zbest file, downsample the skewers ...

Example:
```
quickquasars -i /project/projectdirs/desi/mocks/lya_forest/v1.1/v1.1.0/17/1712/transmission-16-1712.fits --zmin 1.6 --nproc 24 --outdir /project/projectdirs/desi/users/jguy/sim2017/qso-v1.1.0/spectra-16 --downsampling 1. --zbest 
```

This PR is not ready for merging. I am now running Jean-Marc's mocks with this code.
Can @londumas and @andreufont have a look at the code and the results in 
```/project/projectdirs/desi/users/jguy/sim2017/qso-v1.1.0/spectra-16``` ?

### CODE REVIEW

**Comment by moustakas** [comment]
Just for the record: a lot of the core code of generating the noiseless spectra to be simulated---with some features missing---replicates existing code in `desitarget.mock.mockmaker.LYAMaker`.

File: py/desisim/scripts/quickquasars.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        log.info("Resample to a linear wavelength grid (needed by DESI sim.)")
+        qso_wave=np.linspace(args.wmin,args.wmax,int((args.wmax-args.wmin)/args.dwave)+1)
+        qso_flux=np.zeros((tmp_qso_flux.shape[0],qso_wave.size))
+        for q in range(tmp_qso_flux.shape[0]) :
+            qso_flux[q]=np.interp(qso_wave,tmp_qso_wave,tmp_qso_flux[q])
+
+
+        log.info("Apply lya")
+        qso_flux = apply_lya_transmission(qso_wave,qso_flux,trans_wave,transmission)
+
```
Comment (moustakas) [inline py/desisim/scripts/quickquasars.py]: If you want to apply target selection you can do it here with

```
from desitarget.cuts import isQSO_colors
isqso = isQSO_colors(gflux=meta['FLUX_G'], rflux=meta['FLUX_R'], 
    zflux=meta['FLUX_Z'], w1flux=meta['FLUX_W1'], w2flux=meta['FLUX_W2'])
```
and just keep targets with `isqso == 1`.

File: py/desisim/scripts/quickquasars.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        log.info("Resample to a linear wavelength grid (needed by DESI sim.)")
+        qso_wave=np.linspace(args.wmin,args.wmax,int((args.wmax-args.wmin)/args.dwave)+1)
+        qso_flux=np.zeros((tmp_qso_flux.shape[0],qso_wave.size))
+        for q in range(tmp_qso_flux.shape[0]) :
+            qso_flux[q]=np.interp(qso_wave,tmp_qso_wave,tmp_qso_flux[q])
+
+
+        log.info("Apply lya")
+        qso_flux = apply_lya_transmission(qso_wave,qso_flux,trans_wave,transmission)
```
Comment (moustakas) [inline py/desisim/scripts/quickquasars.py]: If you want to do target selection (below) you also need to generate the model spectra over the full wavelength range and then re-synthesize photometry after applying the Lya transmission (+DLAs, etc.) and then resample to the desired output wavelength spacing.  Please see this chunk of code --
https://github.com/desihub/desitarget/blob/master/py/desitarget/mock/mockmaker.py#L1898-L1918

Comment (julienguy) [reply py/desisim/scripts/quickquasars.py]: Thank you John. I will add this target selection part to the code (and add an argument to the script to turn it on/off ).

**Comment by julienguy** [comment]
Ready to merge

**Comment by sbailey** [comment]
Travis is busy today and tests are intermittently timing out.

I agree that this continues the trend of N>>1 ways to simulate spectra with overlapping functionality, but the LyA team is having a great time making end-to-end progress from mocks -> spectra -> correlation functions so I don't want to block that.  We can refactor again someday[*] ...

OK to merge if/when Travis tests pass (I'll be on the road for the next N>>1 hours, so don't necessarily wait for me...)

[*] As Dolores said in Westworld, "*Someday* sounds a lot like the thing people say when they mean *never*."  But maybe...

**Pull request merged**